File: services/order-service/src/routes/health.js
const { Router } = require('express');
const db = require('../db');
const queue = require('../queue');

const router = Router();

// liveness: process is up, no dependency checks
router.get('/healthz', (req, res) => {
  res.json({ status: 'ok' });
});

// readiness: DynamoDB table and SQS queue reachable
router.get('/readyz', async (req, res) => {
  const [dbOk, queueOk] = await Promise.all([db.ping(), queue.ping()]);
  if (dbOk && queueOk) {
    res.json({ status: 'ready' });
  } else {
    res.status(503).json({ status: 'not ready', db: dbOk, queue: queueOk });
  }
});

module.exports = router;

File: services/menu-service/app/routers/health.py
from fastapi import APIRouter, HTTPException, Request

router = APIRouter(tags=["health"])


@router.get("/healthz")
def healthz() -> dict:
    """Liveness: process is up. No dependency checks."""
    return {"status": "ok"}


@router.get("/readyz")
def readyz(request: Request) -> dict:
    """Readiness: DynamoDB table reachable."""
    if request.app.state.repo.ping():
        return {"status": "ready"}
    raise HTTPException(status_code=503, detail="dependencies unavailable")
